src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -265,16 +269,16 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 	 * @async
 	 */
 	async updateTeacher(user, tspTeacher) {
-		const equal =
-			(user.namePrefix === tspTeacher.lehrerTitel || (!user.namePrefix && !tspTeacher.namePrefix)) &&
-			user.firstName === tspTeacher.lehrerVorname &&
-			user.lastName === tspTeacher.lehrerNachname;
-		if (equal) {
-			this.stats.users.teachers.unchanged += 1;
-			return user;
-		}
-
 		try {
```

Comment:
If the error occurres in this block it means that user or tspTeacher are undefined. I would use undefined check here